app/models/vim_performance_state.rb

```diff
@@ -231,6 +221,25 @@ def capture_assoc_ids
     self.assoc_ids = result.presence
   end
 
+  def fetch_assoc_records(resource, assoc)
+    method = if assoc == :vms
+               if resource.kind_of?(EmsCluster)
+                 :all_vms_and_templates
+               elsif resource.kind_of?(Service)
+                 :vms
+               else
+                 :vms_and_templates
+               end
+             else
+               assoc
+             end
+    return false unless resource.respond_to?(method)
```

Comment:
this was a minor change to return something so the parent knows we need to `next`